engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb

```diff
@@ -24,11 +24,12 @@ def path_allowed?(path)
       has_sles11 = @system.products.where(identifier: 'SUSE_SLES').first
       return true if (has_sles11 && (path =~ %r{/12/} || path =~ %r{/12-SP1/}))
 
-      @system.repositories.pluck(:local_path).each do |allowed_path|
-        return true if path =~ /^#{Regexp.escape(allowed_path)}/
-      end
+      all_allowed_paths.find { |allowed_path| path =~ /^#{Regexp.escape(allowed_path)}/ }
```

Comment:
It seems this method is expected to return a boolean, so maybe use `all_allowed_paths.any?` here?